src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -127,8 +127,8 @@ public static SourceText From(Stream stream, Encoding encoding = null, SourceHas
                 throw new InvalidDataException();
             }
 
-            var checksum = CalculateChecksum(stream, checksumAlgorithm);
-            return new StringText(text, encoding, checksum, checksumAlgorithm);
+            Debug.Assert(encoding != null);
+            return new StringText(text, encoding, checksumAlgorithm: checksumAlgorithm);
```

Comment:
I believe the bug is rather in http://source.roslyn.io/#Microsoft.CodeAnalysis/EncodedStringText.cs,e50791c52a80b730,references
where the optimization is incorrectly constructing SourceText.